### DISCUSSION TITLE
Why modifying variables defined in setup only works in dev mode but not in production.

### DISCUSSION THREAD

**Discussion by andreas83** [nuxt/nuxt#17198] [Questions] [answered]
Hello, 
i have a strange behavior, i am able to change a reference defined in the "setup" segment in dev mode but not in production. 
------------------------------
- Operating System: `Linux`
- Node Version:     `v16.13.1`
- Nuxt Version:     `3.0.0-rc.6`
- Package Manager:  `yarn@1.22.10`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

```
<script setup type="ts">
const foo=ref({foo:"bar"});

</script>

<template>
  <div>
    {{foo}}
    <button v-on:click="changeFoo">Change Foo</button>
  </div>
</template>
<script>
import { defineComponent } from '@vue/composition-api'

export default defineComponent({
  setup() {
    
  },
  methods: {
    changeFoo(){
      this.foo.foo="blub";
    }
  }
})
</script>

```

**Result in dev mode (yarn dev)**

Foo becomes blub on click 

**Result while using prod build (yarn build && node .output/server/index.mjs)**

TypeError: Cannot set properties of undefined (setting 'foo')

**Comment by TheAlexLichter** [comment, marked as answer]
Hey! See https://github.com/nuxt/framework/issues/6065 ☺️

**Reply by andreas83** [reply]
thank you!!
